src/common/snippets/include/snippets/generator.hpp

```diff
@@ -42,8 +42,14 @@ class TargetMachine {
     virtual size_t get_lanes() const = 0;
 
     /**
-     * @brief called by generator to all the emittor for a target machine
-     * @return a map by node's type info with callbacks to create an instance of emmitter for corresponding operation type
+    * @brief gets vector of supported element types for execution
+    * @return vector of element types
+    */
+    virtual element::TypeVector get_supported_exec_types() const = 0;
```

Comment:
It's not clear why this is TargetMachine's method, because TargetMachine is designed provide some low-level hardware-specific information. Moreover, you always obtain this info from generator instance, like `m_generator->get_target_machine()->get_supported_exec_types()`, so maybe move it to generator?